plugins/home/src/componentRenderers/ComponentAccordion.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Typography,
+  IconButton,
+  Theme,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import SettingsIcon from '@material-ui/icons/Settings';
+
+import { SettingsModal } from '../components';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  settingsIconButton: {
+    padding: theme.spacing(0, 1, 0, 0),
+  },
+}));
+
+export const ComponentAccordion = ({
+  title,
+  Content,
+  Actions,
+  Settings,
+  ContextProvider,
+  ...childProps
+}: {
+  title: string;
+  Content: React.LazyExoticComponent<(props: any) => JSX.Element>;
```

Comment:
So to whatever extend is reasonable we want to avoid the use of `any` here and other places, but let's not block on it. And ofc skip the lazy exotic component here and in other places as well